MekHQ/src/mekhq/campaign/personnel/death/AbstractDeath.java

```diff
@@ -0,0 +1,291 @@
+import mekhq.MHQConstants;
+import mekhq.MekHQ;
+import mekhq.MekHqXmlUtil;
+import mekhq.campaign.Campaign;
+import mekhq.campaign.CampaignOptions;
+import mekhq.campaign.personnel.Person;
+import mekhq.campaign.personnel.enums.*;
+import org.apache.logging.log4j.LogManager;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.InputStream;
+import java.time.LocalDate;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.ResourceBundle;
+
+public abstract class AbstractDeath {
+    //region Variable Declarations
+    private final RandomDeathMethod method;
+    private Map<AgeGroup, Boolean> enabledAgeGroups;
+    private boolean useRandomClanPersonnelDeath;
+    private boolean useRandomPrisonerDeath;
+    private final boolean enableRandomDeathSuicideCause;
+    private final Map<Gender, Map<TenYearAgeRange, WeightedDoubleMap<PersonnelStatus>>> causes;
+
+    private final ResourceBundle resources = ResourceBundle.getBundle("mekhq.resources.Personnel",
+            MekHQ.getMHQOptions().getLocale(), new EncodeControl());
+    //endregion Variable Declarations
+
+    //region Constructors
+    protected AbstractDeath(final RandomDeathMethod method, final CampaignOptions options) {
+        this.method = method;
+        setEnabledAgeGroups(options.getEnabledRandomDeathAgeGroups());
+        setUseRandomClanPersonnelDeath(options.isUseRandomClanPersonnelDeath());
+        setUseRandomPrisonerDeath(options.isUseRandomPrisonerDeath());
+        this.enableRandomDeathSuicideCause = options.isUseRandomDeathSuicideCause();
+        this.causes = new HashMap<>();
+        initializeCauses();
+    }
+    //endregion Constructors
+
+    //region Getters
+    public RandomDeathMethod getMethod() {
+        return method;
+    }
+
+    public Map<AgeGroup, Boolean> getEnabledAgeGroups() {
+        return enabledAgeGroups;
+    }
+
+    public void setEnabledAgeGroups(final Map<AgeGroup, Boolean> enabledAgeGroups) {
+        this.enabledAgeGroups = enabledAgeGroups;
+    }
+
+    public boolean isUseRandomClanPersonnelDeath() {
+        return useRandomClanPersonnelDeath;
+    }
+
+    public void setUseRandomClanPersonnelDeath(final boolean useRandomClanPersonnelDeath) {
+        this.useRandomClanPersonnelDeath = useRandomClanPersonnelDeath;
+    }
+
+    public boolean isUseRandomPrisonerDeath() {
+        return useRandomPrisonerDeath;
+    }
+
+    public void setUseRandomPrisonerDeath(final boolean useRandomPrisonerDeath) {
+        this.useRandomPrisonerDeath = useRandomPrisonerDeath;
+    }
+
+    public boolean isEnableRandomDeathSuicideCause() {
+        return enableRandomDeathSuicideCause;
+    }
+
+    public Map<Gender, Map<TenYearAgeRange, WeightedDoubleMap<PersonnelStatus>>> getCauses() {
+        return causes;
+    }
+    //endregion Getters
+
+    /**
+     * This is used to determine if a person can divorce.
```

Comment:
Though some people may view it that way, divorce != death

Reply:
I copied the base design from divorce during modernization, must have missed this comment.